common/data/src/main/java/org/thingsboard/server/common/data/tenant/profile/DefaultTenantProfileConfiguration.java

```diff
@@ -197,4 +204,10 @@ public TenantProfileType getType() {
     public int getMaxRuleNodeExecsPerMessage() {
         return maxRuleNodeExecutionsPerMessage;
     }
+
+    @Override
+    public int getMaxDebugModeDurationMinutes(int systemMaxDebugModeDurationMinutes) {
+        return maxDebugModeDurationMinutes > 0 ? maxDebugModeDurationMinutes : systemMaxDebugModeDurationMinutes > 0 ?
```

Comment:
I have received the brain damage while parsing this. Please simplify with if statements. Add note that profile value has more priority then system. Rename `systemMaxDebugModeDurationMinutes` to `systemDefaultDebugModeDurationMinutes`

Reply:
We can discuss this, but it looks like spaghetti code to me. why do we pass `systemMaxDebugModeDurationMinutes` into the Configuration object to calculate the `maxDebugModeDurationMinutes` which is a field in the configuration?